rclcpp_examples/src/parameters/parameter_events_async.cpp

```diff
@@ -16,20 +16,36 @@
 
 #include "rclcpp/rclcpp.hpp"
 
-void on_parameter_event(const rcl_interfaces::msg::ParameterEvent::SharedPtr event)
+void on_parameter_event(
+  const std::set<std::string> & expected_new_parameters,
+  const std::set<std::string> & expected_changed_parameters,
+  const std::set<std::string> & expected_deleted_parameters,
+  std::set<std::string> & received_new_parameters,
+  std::set<std::string> & received_changed_parameters,
+  std::set<std::string> & received_deleted_parameters,
+  const rcl_interfaces::msg::ParameterEvent::SharedPtr event)
 {
   // TODO(wjwwood): The message should have an operator<<, which would replace all of this.
   std::cout << "Parameter event:" << std::endl << " new parameters:" << std::endl;
   for (auto & new_parameter : event->new_parameters) {
     std::cout << "  " << new_parameter.name << std::endl;
+    received_new_parameters.insert(new_parameter.name);
   }
   std::cout << " changed parameters:" << std::endl;
   for (auto & changed_parameter : event->changed_parameters) {
     std::cout << "  " << changed_parameter.name << std::endl;
+    received_changed_parameters.insert(changed_parameter.name);
   }
   std::cout << " deleted parameters:" << std::endl;
   for (auto & deleted_parameter : event->deleted_parameters) {
     std::cout << "  " << deleted_parameter.name << std::endl;
+    received_deleted_parameters.insert(deleted_parameter.name);
+  }
+
+  if (expected_new_parameters == received_new_parameters &&
+    expected_changed_parameters == received_changed_parameters &&
+    expected_deleted_parameters == received_deleted_parameters) {
```

Comment:
One option would be:

    if (
      cond &&
      cond
    ) {
      ...
    }